pkg/pipeline/stream/acker.go

```diff
@@ -188,6 +188,12 @@ func (n *AckerNode) ForgetAndDrop(msg *Message) {
 // Wait can be used to wait for the count of outstanding acks to drop to 0 or
 // the context gets canceled.
 func (n *AckerNode) Wait(ctx context.Context) {
+	// in case ExpectAck wasn't called we call it here
+	n.startOnce.Do(func() {
+		n.init()
+		close(n.start)
+	})
```

Comment:
The issue was the node was stuck in the select statement in line 77, now we ensure it continues on there even if no messages flowed through the pipeline.